aioshelly/rpc_device.py

```diff
@@ -88,13 +88,12 @@ async def initialize(self) -> None:
         try:
             self.shelly = await get_info(self.aiohttp_session, self.options.ip_address)
 
-            if self.options.auth or not self.requires_auth:
-                await self._wsrpc.connect(self.aiohttp_session)
-                await asyncio.gather(
-                    self.update_device_info(),
-                    self.update_config(),
-                    self.update_status(),
-                )
+            await self._wsrpc.connect(self.aiohttp_session, self.options)
+            await asyncio.gather(
```

Comment:
Instead of doing the auth dance in `WsRPC.connect()` I would reorganize this flow instead:
 * fetch device info
 * check if `auth_en`, if yes:
   * password is unknown -- error out.
   * password is known -- make sure WsRPC knows about it, possibly make it send the `auth` data without getting a 401 error first
   * proffit